src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Azure/Functions/AzureFunctionExecutorTryExecuteAsyncIntegration.cs

```diff
@@ -0,0 +1,61 @@
+// <copyright file="AzureFunctionExecutorTryExecuteAsyncIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Threading;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Azure.Functions
+{
+    /// <summary>
+    /// Azure Function calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "Microsoft.Azure.WebJobs.Host",
+        TypeName = "Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor",
+        MethodName = "TryExecuteAsync",
+        ReturnTypeName = "System.Threading.Tasks.Task`1[Microsoft.Azure.WebJobs.Host.Executors.IDelayedException]",
+        ParameterTypeNames = new[] { "Microsoft.Azure.WebJobs.Host.Executors.IFunctionInstance", "System.Threading.CancellationToken" },
+        MinimumVersion = "3.0.0",
+        MaximumVersion = "3.*.*",
+        IntegrationName = IntegrationName)]
+    public class AzureFunctionExecutorTryExecuteAsyncIntegration
```

Comment:
[branding nitpick] Consider renaming to `AzureFunctionsExecutorTryExecuteAsyncIntegration` since the product is called "Azure Functions" (in plural)